src/ProjectContext.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright 2021 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\CloudFunctions;
+
+use Google\Cloud\Storage\StreamWrapper;
+use Google\Cloud\Storage\StorageClient;
+use RuntimeException;
+
+class ProjectContext
+{
+    public function locateAutoloadFile(): ?string
+    {
+        /**
+         * Determine the autoload file to load.
+         */
+        if (file_exists($vendored = __DIR__ . '/../../../autoload.php')) {
```

Comment:
Should we make this more fault-tolerant, or can we depend on these hard-coded paths remaining constant?

Reply:
How do you suggest making this more fault tolerant?

As long as we don't rename this file, I do not believe there will be an issue. And we have tests that will fail if we rename the file.